fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/StorageUtil.java

```diff
@@ -86,4 +86,50 @@
 
 		return totalMoved;
 	}
+
+	/**
+	 * Attempt to find a resource stored in the passed storage.
+	 * @param storage The storage to inspect.
+	 * @param transaction The current transaction, or {@code null} if a transaction should be opened for this query.
+	 * @param <T> The type of the stored resources.
+	 * @return A non-empty resource stored in the storage, or {@code null} if none could be found.
+	 */
+	@Nullable
+	public static <T> T findStoredResource(Storage<T> storage, @Nullable Transaction transaction) {
+		if (transaction == null) {
+			transaction = Transaction.openOuter();
```

Comment:
I need to put this in a try block...

Reply:
I suggest moving the loop into another method and you decide to whether call the method in a try block or not depending on the state of the given transaction

Reply:
Yeah it's a little bit cleaner :)

Reply:
Done, feel free to review the rest of the PR as well :)